dask/blockwise.py

```diff
@@ -381,23 +432,11 @@ def __dask_distributed_pack__(
         # All blockwise tasks will depend on the futures in `indices`
         global_dependencies = {stringify(f.key) for f in indices_unpacked_futures}
 
-        # Handle `io_deps` serialization.
-        # If `io_deps[<collection_key>]` is just a dict, we rely
-        # entirely on msgpack.  It is up to the `Blockwise` layer to
-        # ensure that all arguments are msgpack serializable. To enable
-        # more control over serialization, a `BlockwiseIODeps` mapping
-        # subclass can be defined with the necessary
-        # `__dask_distributed_{pack,unpack}__` methods.
+        # Handle `io_deps` serialization. Assume each element
+        # is a `BlockwiseDep`-based object.
         packed_io_deps = {}
         for name, input_map in self.io_deps.items():
-            if isinstance(input_map, tuple):
-                # Use the `__dask_distributed_pack__` definition for the
-                # specified `BlockwiseIODeps` subclass
-                module_name, attr_name = input_map[0].rsplit(".", 1)
-                io_dep_map = getattr(import_allowed_module(module_name), attr_name)
-                packed_io_deps[name] = io_dep_map.__dask_distributed_pack__(*input_map)
-            else:
-                packed_io_deps[name] = input_map
+            packed_io_deps[name] = input_map.__dask_distributed_pack__()
```

Comment:
Sounds good to me. We will certainly want something like this for culling. Does it seem reasonable to just pass in `self.output_blocks` (rather than having any special array-like object in Blockwise)?